qiskit/aqua/operators/state_fns/operator_state_fn.py

```diff
@@ -34,7 +34,7 @@ class OperatorStateFn(StateFn):
 
     # TODO allow normalization somehow?
     def __init__(self,
-                 primitive: Union[OperatorBase] = None,
+                 primitive: Optional[OperatorBase] = None,
```

Comment:
Same with this here I think we should delete the `= None` instead of making it optional. My recollection is the base class primitive op has some type checking on primitive and will fail if this is really ever set as None.